File: Projet_SHAITA_DRIOUCHE_INFO0503/serveur_PHP/package_php/enum/OrigineEnergie.php
<?php

class OrigineEnergie {
    // Attributs
    public static $AUCUNE = null;
    public static $FRANCE = null;
    public static $ALLEMAGNE = null;
    public static $ESPAGNE = null;
    public static $CHINE = null;
    public static $ITALIE = null;
    public static $ROYAUMEUNI = null;
    public static $PAYSBAS = null;
    public static $PORTUGAL = null;
    public static $SUEDE = null;
    public static $FINLANDE = null;
    public static $ALGERIE = null;
    public static $MAROC = null;
    public static $TUNISIE = null;
    public static $TURQUIE = null;

    private $nom;

    // Constructeur
    public function __construct($nom){
        $this->nom = $nom;
    }

    public function toString() : String{ return $this->nom; }

    
}

OrigineEnergie::$AUCUNE = new OrigineEnergie("Aucune");
OrigineEnergie::$FRANCE = new OrigineEnergie("France");
OrigineEnergie::$ALLEMAGNE = new OrigineEnergie("Allemagne");
OrigineEnergie::$ESPAGNE = new OrigineEnergie("Espagne");
OrigineEnergie::$CHINE = new OrigineEnergie("Chine");
OrigineEnergie::$ITALIE = new OrigineEnergie("Italie");
OrigineEnergie::$ROYAUMEUNI = new OrigineEnergie("Royaume-Uni");
OrigineEnergie::$PAYSBAS = new OrigineEnergie("Pays-Bas");
OrigineEnergie::$PORTUGAL = new OrigineEnergie("Portugal");
OrigineEnergie::$SUEDE = new OrigineEnergie("Suede");
OrigineEnergie::$FINLANDE = new OrigineEnergie("Finlande");
OrigineEnergie::$ALGERIE = new OrigineEnergie("Algerie");
OrigineEnergie::$MAROC = new OrigineEnergie("Maroc");
OrigineEnergie::$TUNISIE = new OrigineEnergie("Tunisie");
OrigineEnergie::$TURQUIE = new OrigineEnergie("Turquie");


$tabOrigineEnergie = array(OrigineEnergie::$AUCUNE,OrigineEnergie::$FRANCE, OrigineEnergie::$ALLEMAGNE, OrigineEnergie::$ESPAGNE, OrigineEnergie::$CHINE, OrigineEnergie::$ITALIE, OrigineEnergie::$ROYAUMEUNI, OrigineEnergie::$PAYSBAS, OrigineEnergie::$PORTUGAL, OrigineEnergie::$SUEDE, OrigineEnergie::$FINLANDE, OrigineEnergie::$ALGERIE, OrigineEnergie::$MAROC, OrigineEnergie::$TUNISIE, OrigineEnergie::$TURQUIE);

?>

File: Projet_SHAITA_DRIOUCHE_INFO0503/serveur_JAVA/src/package_java/enums/OrigineEnergie.java
package package_java.enums;

public enum OrigineEnergie {
    AUCUNE("Aucune"),
    FRANCE("France"),
    ALLEMAGNE("Allemagne"),
    ESPAGNE("Espagne"),
    CHINE("Chine"),
    ITALIE("Italie"),
    ROYAUMEUNI("Royaume-Uni"),
    PAYSBAS("Pays-Bas"),
    PORTUGAL("Portugal"),
    SUEDE("Suede"),
    FINLANDE("Finlande"),
    ALGERIE("Algerie"),
    MAROC("Maroc"),
    TUNISIE("Tunisie"),
    TURQUIE("Turquie");

    private String nom;

    OrigineEnergie(String nom) {
        this.nom = nom;
    }

    public String toString() {
        return nom;
    }

    public static OrigineEnergie getValue(String s) {
        for (OrigineEnergie c : OrigineEnergie.values()) {
            if (c.nom.equals(s))
                return c;
        }
        return null;
    }

}
